### ISSUE TITLE
[BUG] AttributeError: 'list' object has no attribute 'float'

### ISSUE THREAD

**Issue by tamararobin1982** [PennyLaneAI/pennylane#4124]
### Expected behavior

Is need give float, but give list so error
```
@qml.qnode(dev, interface='torch')
def q_net(q_in, q_weights_flat):
        
        # Reshape weights
        q_weights = q_weights_flat.reshape(max_layers, n_qubits)
        
        # Start from state |+> , unbiased w.r.t. |0> and |1>
        H_layer(n_qubits)   
        
        # Embed features in the quantum node
        RY_layer(q_in)      
       
        # Sequence of trainable variational layers
        for k in range(q_depth):
            entangling_layer(n_qubits)
            RY_layer(q_weights[k + 1])

        # Expectation values in the Z basis
        retVal= [qml.expval(qml.PauliZ(j)) for j in range(n_qubits)]
        print((retVal))
        return retVal
```

### Actual behavior

Not make return float :
` q_out_elem = q_net(elem,self.q_params).float().unsqueeze(0)`

```
Cell In[63], line 17, in Quantumnet.forward(self, input_features)
     15 for elem in q_in:
     16     type(q_in)
---> 17     q_out_elem = q_net(elem,self.q_params).float().unsqueeze(0)
     18     q_out = torch.cat((q_out, q_out_elem))
     19 return self.post_net(q_out)

AttributeError: 'list' object has no attribute 'float'
```

### Source code

```shell
class Quantumnet(nn.Module):
        def __init__(self):
            super().__init__()
            self.pre_net = nn.Linear(512, n_qubits)
            self.q_params = nn.Parameter(q_delta * torch.randn(max_layers * n_qubits))
            self.post_net = nn.Linear(n_qubits, 2)

        def forward(self, input_features):
            pre_out = self.pre_net(input_features) 
            q_in = torch.tanh(pre_out) * np.pi / 2.0   
            
            # Apply the quantum circuit to each element of the batch and append to q_out
            q_out = torch.Tensor(0, n_qubits)
            q_out = q_out.to(device)
            for elem in q_in:
                type(q_in)
                q_out_elem = q_net(elem,self.q_params).float().unsqueeze(0)
                q_out = torch.cat((q_out, q_out_elem))
            return self.post_net(q_out)
```

### System information

```shell
Name: PennyLane
Version: 0.30.0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /home/.local/lib/python3.8/site-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, rustworkx, scipy, semantic-version, toml
Required-by: PennyLane-Lightning

Platform info:           Linux-5.10.16.3-microsoft-standard-WSL2-x86_64-with-glibc2.29
Python version:          3.8.10
Numpy version:           1.23.5
Scipy version:           1.9.1
Installed devices:
- default.gaussian (PennyLane-0.30.0)
- default.mixed (PennyLane-0.30.0)
- default.qubit (PennyLane-0.30.0)
- default.qubit.autograd (PennyLane-0.30.0)
- default.qubit.jax (PennyLane-0.30.0)
- default.qubit.tf (PennyLane-0.30.0)
- default.qubit.torch (PennyLane-0.30.0)
- default.qutrit (PennyLane-0.30.0)
- null.qubit (PennyLane-0.30.0)
- lightning.qubit (PennyLane-Lightning-0.30.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by rmoyard** [comment]
Hi @tamararobin1982, since the last pennylane release we updated the return type of qnodes. Your qnode returns a list (where previously it was an array) and therefore you cannot apply `.float` on it.

You should update this line:

```
q_out_elem = q_net(elem,self.q_params).float().unsqueeze(0)
```
and simply stack the results that are in the list.

Let us know if it works!

**Comment by rmoyard** [comment]
I think something like that should work but without more details it is complicated to solve this issue.

```
 q_out_elem_lst = q_net(elem,self.q_params)
 q_out = toch.hstack(q_out_elem_lst)
```

**Comment by josh146** [comment]
Hi @tamararobin1982! I'll close this issue for now, but feel free to open a new issue if you come across any other problems or bugs :)
